src/main/java/com/musala/atmosphere/commons/websocket/util/UiElementSelectorDeserializer.java

```diff
@@ -0,0 +1,83 @@
+package com.musala.atmosphere.commons.websocket.util;
+
+import java.lang.reflect.Type;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.apache.log4j.Logger;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.musala.atmosphere.commons.geometry.Bounds;
+import com.musala.atmosphere.commons.ui.selector.CssAttribute;
+import com.musala.atmosphere.commons.ui.selector.UiElementSelectionOption;
+import com.musala.atmosphere.commons.ui.selector.UiElementSelector;
+
+/**
+ * Deserializer for the {@link UiElementSelector} object.
+ *
+ * @author dimcho.nedev
+ *
+ */
+public class UiElementSelectorDeserializer implements JsonDeserializer<UiElementSelector> {
+    private final static Logger LOGGER = Logger.getLogger(UiElementSelectorDeserializer.class);
+
+    public static final String KEY = "mKey";
+
+    public static final String VALUE = "mValue";
+
+    @Override
+    public UiElementSelector deserialize(JsonElement json, Type type, JsonDeserializationContext context)
+            throws JsonParseException {
+        UiElementSelector selector = new UiElementSelector();
+
+        JsonObject jsonObject = json.getAsJsonObject();
+        JsonElement attributeProjectionMap = jsonObject.get("attributeProjectionMap");
+
+        if (attributeProjectionMap.equals(new JsonObject())) {
+            // empty attribute map: {}
+            return selector;
+        }
+        LOGGER.debug(String.format("attributeProjectionMap: %s", attributeProjectionMap));
```

Comment:
Later I'll change the log level globally from the log4j.properties file but I find it very useful for now